File: CSharp/DP.Domain/Samples/Mediator/OptionColleague.cs
namespace DP.Domain.Samples.Mediator
{
    /// <summary>
    /// 選擇權評分模型
    /// </summary>
    public class OptionColleague : IColleague
    {
        public string Prod { get; set; } = "期貨/選擇權";
        public IMediator Mediator { get; set; }

        public OptionColleague(IMediator mediator=null)
        {
            this.Mediator = mediator;
        }
        public decimal Score()
        {
            //Implement the real score model here.
            return 10;
        }
    }
}

File: Python/Samples/Mediator/OptionColleague.py
from Colleague import Colleague
from Mediator import Mediator

class OptionColleague(Colleague):
    """選擇權評分模型"""
    
    _mediator = None

    def __init__(self):
        self.prod = "期貨/選擇權"

    @property
    def mediator(self):
        return self._mediator

    @mediator.setter
    def mediator(self, val=Mediator):
        self._mediator = val

    def score(self):
        # Implement the real score model here.
        return 10

